File: renderer/src/stores/works.ts
import { create } from 'zustand';
import { api } from '../services/engine';
import type {
  Work,
  WorkDetail,
  WorksListResponse,
  WorksQueryParams,
} from '@shared/ipc-types';

interface WorksState {
  works: Work[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
  loading: boolean;
  error: string | null;
  query: WorksQueryParams;
  loadWorks: (params?: Partial<WorksQueryParams>) => Promise<void>;
  getWork: (id: string) => Promise<WorkDetail | null>;
  renameWork: (id: string, name: string) => Promise<boolean>;
  deleteWork: (id: string) => Promise<boolean>;
  batchDelete: (ids: string[]) => Promise<boolean>;
  clearAll: () => Promise<boolean>;
}

export const useWorksStore = create<WorksState>((set, get) => ({
  works: [],
  total: 0,
  page: 1,
  pageSize: 12,
  totalPages: 0,
  loading: false,
  error: null,
  query: {},

  loadWorks: async (params) => {
    const query = { ...get().query, ...params };
    set({ loading: true, error: null, query });

    const searchParams = new URLSearchParams();
    if (query.search) searchParams.set('search', query.search);
    if (query.aspect_ratio) searchParams.set('aspect_ratio', query.aspect_ratio);
    if (query.date_range) searchParams.set('date_range', query.date_range);
    if (query.sort) searchParams.set('sort', query.sort);
    searchParams.set('page', String(query.page ?? 1));
    searchParams.set('page_size', String(query.page_size ?? 12));

    const res = await api.get<WorksListResponse>(`/api/works?${searchParams}`);
    if (res.success && res.data) {
      set({
        works: res.data.items,
        total: res.data.total,
        page: res.data.page,
        pageSize: res.data.page_size,
        totalPages: res.data.total_pages,
        loading: false,
      });
    } else {
      set({ loading: false, error: res.error?.message ?? '加载作品失败' });
    }
  },

  getWork: async (id) => {
    const res = await api.get<WorkDetail>(`/api/works/${id}`);
    return res.success && res.data ? res.data : null;
  },

  renameWork: async (id, name) => {
    const res = await api.patch(`/api/works/${id}`, { name });
    if (res.success) {
      await get().loadWorks();
      return true;
    }
    return false;
  },

  deleteWork: async (id) => {
    const res = await api.delete(`/api/works/${id}`);
    if (res.success) {
      await get().loadWorks();
      return true;
    }
    return false;
  },

  batchDelete: async (ids) => {
    const res = await api.delete('/api/works', { ids });
    if (res.success) {
      await get().loadWorks();
      return true;
    }
    return false;
  },

  clearAll: async () => {
    const res = await api.delete('/api/works/all', { confirm: true });
    if (res.success) {
      await get().loadWorks();
      return true;
    }
    return false;
  },
}));

File: python-engine/src/api/routes/works.py
"""Works library CRUD routes."""

from fastapi import APIRouter

from src.storage.works_repo import works_repo

router = APIRouter(tags=["works"])


@router.get("/works")
async def list_works(
    search: str = "",
    aspect_ratio: str = "",
    date_range: str = "",
    date_from: str = "",
    date_to: str = "",
    sort: str = "created_at_desc",
    page: int = 1,
    page_size: int = 12,
):
    data = works_repo.list(
        search=search,
        aspect_ratio=aspect_ratio,
        date_range=date_range,
        date_from=date_from,
        date_to=date_to,
        sort=sort,
        page=page,
        page_size=page_size,
    )
    return {"success": True, "data": data}


@router.get("/works/{work_id}")
async def get_work(work_id: str):
    work = works_repo.get_by_id(work_id)
    if not work:
        return {"success": False, "error": {"code": "NOT_FOUND", "message": "作品不存在"}}
    return {"success": True, "data": work}


@router.patch("/works/{work_id}")
async def rename_work(work_id: str, body: dict):
    name = body.get("name", "").strip()
    if not name:
        return {"success": False, "error": {"code": "INVALID_SCRIPT", "message": "名称不能为空"}}
    work = works_repo.rename(work_id, name)
    if not work:
        return {"success": False, "error": {"code": "NOT_FOUND", "message": "作品不存在"}}
    return {"success": True, "data": work}


@router.delete("/works/{work_id}")
async def delete_work(work_id: str):
    ok = works_repo.delete(work_id)
    if not ok:
        return {"success": False, "error": {"code": "NOT_FOUND", "message": "作品不存在"}}
    return {"success": True, "data": {"deleted": True}}


@router.delete("/works")
async def batch_delete_works(body: dict):
    ids = body.get("ids", [])
    count = works_repo.batch_delete(ids)
    return {"success": True, "data": {"deleted_count": count}}


@router.delete("/works/all")
async def clear_all_works(body: dict):
    if not body.get("confirm"):
        return {"success": False, "error": {"code": "INTERNAL_ERROR", "message": "需要 confirm: true"}}
    count = works_repo.clear_all()
    return {"success": True, "data": {"deleted_count": count}}
